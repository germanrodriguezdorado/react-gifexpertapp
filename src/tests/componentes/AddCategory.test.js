import { shallow } from "enzyme";
import AddCategory from "../../componentes/AddCategory";
import "@testing-library/jest-dom";

describe('Pruebas correspondientes al AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });


    test('El componente se debe renderizar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find("input");
        const value = "Hola Mundo!";
        input.simulate("change", { 
            target: { 
                value: value
            } 
        });

    });

    test('No debe postear la info con submit', () => {
        wrapper.find("form").simulate("submit", { preventDefault: () => {} })
        expect(setCategories).not.toHaveBeenCalled();
    });


    test("Debe de llamar al setCategories y limpiar la caja de texto", () => {

        const input = wrapper.find("input");
        const value = "Captain America";
        input.simulate("change", {
            target: {
                value: value
            }
        });

        const form = wrapper.find("form");
        form.simulate("submit", {
            preventDefault: () => {}
        });

        expect(setCategories).toHaveBeenCalled();
        expect(input.prop("value")).toBe("");

    });
    
    
    

});
