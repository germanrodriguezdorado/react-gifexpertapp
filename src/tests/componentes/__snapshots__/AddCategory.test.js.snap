// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas correspondientes al AddCategory El componente se debe renderizar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
